src/System.Reflection.Metadata/specs/PE-COFF.md

```diff
@@ -65,3 +65,43 @@ UncompressedSize shall be greater than 0. Other values are reserved for future u
 The Major version specified in the entry indicates the version of the Portable PDB format. The Minor version indicates the version of the Embedded Portable PDB data format.
 
 The value of Stamp field in the entry shall be 0.
+
+### PDB Checksum Debug Directory Entry (type 18)
+
+Stores crypto hash of the content of the symbol file the PE/COFF file was built with.
+
+The hash can be used to validate that a given PDB file was built with the PE/COFF file and not altered in any way. 
+
+More than one entry can be present, in case multiple PDBs were produced during the build of the PE/COFF file (e.g. private and public symbols). 
+
+*Version Major=0x0001, Minor=0x0000* of the entry data format is following:
+
+| Offset            | Size          | Field          | Description                                                             |
+|:------------------|:--------------|:---------------|-------------------------------------------------------------------------|
+| 0                 | 4             | Signature      | 0x50, 0x44, 0x42, 0x43 (ASCII string: "PDBC")                           |
+| 4                 | AlgNameLength | AlgorithmName  | Zero-terminated ASCII string. The name of the crypho hash algorithm.    |
```

Comment:
I think I would suggest that we have a field for the size of either `Checksum` or the size of the entire record so that a tool can traverse the structure even if it hits a checksum algorithm that it doesn't understand.

Reply:
The size of the data record is already in the debug table.